tutorials/crunch-aggregate-sessions.md

```diff
@@ -0,0 +1,296 @@
+---
+layout: page
+title: Aggregating Records with Crunch
+---
+
+## Purpose
+
+The Apache Crunch Java library provides a framework for writing, testing, and running applications in MapReduce pipelines.
+
+This tutorial demonstrates how you use Kite's `CrunchDatasets` helper methods to configure source and target datasets for Crunch operations that aggregate individual events into work sessions.
+
+### Prerequisites
+
+* A VM or cluster with CDH installed. See [Preparing the Virtual Machine][vm].
+* An [events dataset][events] populated with [sample records][gen-event].
+
+[vm]:{{site.baseurl}}/tutorials/preparing-the-vm.html
+[events]:{{site.baseurl}}/tutorials/create-events-dataset.html
+[gen-event]:{{site.baseurl}}/tutorials/generate-events.html
+
+## Creating the sessions.avsc Schema File
+
+The `session.avsc` schema defines the fields and data types in the `sessions` dataset.
+
+### session.avsc
+
+```JSON
+{
+  "name": "Session",
+  "namespace": "org.kitesdk.examples.demo.event",
+  "type": "record",
+  "doc": "A session derived from a standard event log.",
+  "fields": [
+    {
+      "name": "user_id",
+      "type": "long",
+      "doc": "A unique identifier for the user. Required."
+    },
+    {
+      "name": "session_id",
+      "type": "string",
+      "doc": "A unique identifier for the session. Required."
+    },
+    {
+      "name": "ip",
+      "type": "string",
+      "doc": "The IP address of the host where the event originated. Required."
+    },
+    {
+      "name": "start_timestamp",
+      "type": "long",
+      "doc": "The point in time when the session started, represented as the number of milliseconds since January 1, 1970, 00:00:00 GMT. Required."
+    },
+    {
+      "name": "duration",
+      "type": "long",
+      "doc": "The duration of the session in milliseconds. Required."
+    },
+    {
+      "name": "session_event_count",
+      "type": "int",
+      "doc": "The number of events that occurred in the session. Required."
+    }
+  ]
+}
+```
+
+Save the schema definition as `session.avsc`.
+
+## Creating the Sessions Dataset Using the Kite CLI
+
+Create the `sessions` dataset using the default Hive scheme.
+
+To create the sessions dataset:
+
+1. Open a terminal window.
+1. Use the `create` command to create the dataset. This example assumes that the schema file is in your home directory. Change the path to `session.avsc` if you stored it in another directory.
+
+```
+kite-dataset create sessions --schema ~/session.avsc
```

Comment:
This can assume the user hasn't changed directories, so there's no need for the `~/` part of the path or discussion about location.